### PR TITLE
Fix a number of tests

### PR DESCRIPTION
**Pull request by dragon512**
updated return code based on fixes in new autest drop
clean up tsn and double test to work better ( still needs more work )
change return code test as we can now shut doiwn better

### CODE REVIEW

File: tests/gold_tests/autest-site/init.cli.ext [hunk 1]
```diff
@@ -23,7 +23,7 @@ if sys.version_info < (3, 5, 0):
     host.WriteError(
         "You need python 3.5 or later to run these tests\n", show_stack=False)
 
-if AuTestVersion() < "1.3.5":
+if AuTestVersion() < "1.4.0":
     host.WriteError(
         "Tests need AuTest version 1.3.4 or better\n Please update AuTest:\n  pip install --upgrade autest\n", show_stack=False)
```
Comment (d2r) [inline tests/gold_tests/autest-site/init.cli.ext]: Should this say we need version `1.4.0` or better?

**Comment by dragon512** [comment]
[approve ci autest]

**Comment by dragon512** [comment]
[approve ci autest]

**Comment by dragon512** [comment]
[approve ci autest]

**Comment by dragon512** [comment]
[approve ci autest]
[approve ci autest]

**Comment by dragon512** [comment]
[approve ci autest]

**Pull request merged**